### PR TITLE
Add page for channel search

### PR DESCRIPTION
**Pull request by noisecapella**
#### Pre-Flight checklist

- [ ] Screenshots and design review for any changes that affect layout or styling
  - [ ] Desktop screenshots 
  - [ ] Mobile width screenshots 
  - [ ] tag @ferdi or @pdpinch for review  
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested
- [ ] Settings
  - [ ] New settings are documented and present in `app.json`
  - [x] New settings have reasonable development defaults, if applicable

#### What are the relevant tickets?
Part of #1316 

#### What's this PR do?
Implements channel search

#### How should this be manually tested?
Run `./manage.py recreate_index` to ensure the index exists then run a search on the channel. You should be able to search the title and text for a post and only the text for a comment. You should never see any profile results and the search filter should only include options for comments or posts. Selecting 'All' should only show some combination of the two, no profile results.

### CODE REVIEW

File: search/api.py [hunk 1]
```diff
@@ -90,5 +90,8 @@ def execute_search(*, user, query):
     search = Search(index=index, using=get_conn())
     search.update_from_dict(query)
     channel_names = [channel.display_name for channel in Api(user).list_channels()]
-    search = search.filter("terms", channel_name=channel_names)
+    channels_filter = Q("terms", author_channel_membership=channel_names) | Q(
```
Comment (mbertrand) [inline search/api.py]: Is dealing with public channels going to be in another PR?  This line will prevent searching in any channel that the user is not subscribed to (ie public channels, or conceivably a private channel which the user is a contributor/moderator on but not a subscriber?).

Comment (noisecapella) [reply search/api.py]: I'm going to change this query to restrict to only the channel being viewed, or no channels if the user does not have permission to view that channel.

Comment (noisecapella) [reply search/api.py]: Actually I changed it to use the models for contributor and moderator membership. It would be a significant amount of work to filter only on the user and since this is still behind a feature flag I think we can have a PR to add the public channel part of it before it is open to the public

**Comment by noisecapella** [comment]
Ok, I think that's finally all of the initial feedback

File: search/api.py [hunk 2]
```diff
@@ -84,11 +91,18 @@ def execute_search(*, user, query):
...
     search.update_from_dict(query)
-    channel_names = [channel.display_name for channel in Api(user).list_channels()]
-    search = search.filter("terms", channel_name=channel_names)
+    channel_names = list(
+        ChannelGroupRole.objects.filter(
+            group__user=user, role__in=(ROLE_CONTRIBUTORS, ROLE_MODERATORS)
+        )
+        .values_list("channel__name", flat=True)
+        .distinct()
+    )
```
Comment (mbertrand) [inline search/api.py]: This will prevent subscribers of restricted channels from getting search results (checkbox 2 in #1461).  Using `list(profiles.api.get_channels(user))` will include the channel names that the user subscribes to.

Comment (noisecapella) [reply search/api.py]: I think the only way we can support that is to add the `channel_type` information to the `Channel` model and then update the filter to permit all searches to those channels. I'm going to do that in the next PR and then update this logic to fix the filtering

Comment (mbertrand) [reply search/api.py]: OK, I tried `list(profiles.api.get_channels(user))` and that does work for searching within a restricted channel when logged in as a subscribed user, but I'm fine with solving it another way in a future PR.
